core/server/src/main/java/alluxio/master/block/BlockMaster.java

```diff
@@ -360,30 +367,56 @@ public long getNewContainerId() {
    */
   public void commitBlock(long workerId, long usedBytesOnTier, String tierAlias, long blockId,
       long length) {
-    LOG.debug("Commit block from worker: {}",
-        FormatUtils.parametersToString(workerId, usedBytesOnTier, blockId, length));
-    synchronized (mBlocks) {
-      synchronized (mWorkers) {
-        MasterWorkerInfo workerInfo = mWorkers.getFirstByField(mIdIndex, workerId);
-        workerInfo.addBlock(blockId);
-        workerInfo.updateUsedBytes(tierAlias, usedBytesOnTier);
-        workerInfo.updateLastUpdatedTimeMs();
+    LOG.debug("Commit block from workerId: {}, usedBytesOnTier: {}, blockId: {}, length: {}",
+        workerId, usedBytesOnTier, blockId, length);
+
+    long counter = -1;
 
+    MasterWorkerInfo workerInfo = mWorkers.getFirstByField(mIdIndex, workerId);
+    // Lock the worker metadata first.
+    synchronized (workerInfo) {
+      // Loop until block metadata is successfully locked.
```

Comment:
same here, do we need to check if `workerInfo` still belongs to `mWorkers` after locking?